pkg/controller/publisher.go

```diff
@@ -193,6 +192,16 @@ func (s *SCSIPublisher) addTargetsInfoToPublishContext(
 		publishContext[fmt.Sprintf("%s%d", common.PublishContextNVMEFCPortalsPrefix, i)] = t.Portal
 		publishContext[fmt.Sprintf("%s%d", common.PublishContextNVMEFCTargetsPrefix, i)] = t.Target
 	}
+
+	nvmetcpTargetInfo, err := common.GetNVMETCPTargetsInfoFromStorage(client, volumeApplianceID)
+	if err != nil {
+		log.Error("error unable to get NVMeTCP targets from array", err)
+	}
+	for i, t := range nvmetcpTargetInfo {
+		publishContext[fmt.Sprintf("%s%d", common.PublishContextNVMETCPPortalsPrefix, i)] = t.Portal
+		publishContext[fmt.Sprintf("%s%d", common.PublishContextNVMETCPTargetsPrefix, i)] = t.Target
+	}
+
 	// If the system is not capable of any protocol, then we will through the error
 	if len(iscsiTargetsInfo) == 0 && len(fcTargetsInfo) == 0 && len(nvmefcTargetInfo) == 0 {
```

Comment:
@delldubey Any need to extend this check by adding len(nvmetcpTargetInfo) == 0? Maybe NO since if len(nvmetcpTargetInfo) == 0 then len(iscsiTargetsInfo) must be 0 right? If not, please make sure to add this too.

Reply:
+1
I think this is required.

Reply:
Thanks @delldubey